### PR TITLE
feat(css): parse css selectors (type, attr, nesting)

### PR DESCRIPTION
**Pull request by neuodev**
## Current State 

Parser can now parse all these selectors 

```css
            /* Links that start with "https://" and end in ".org" */
            a[href^="https://"][href$=".org"] {
                color: green;
            } 

            /* Links that end in ".org" */
            a[href$=".org"] {
                color: red;
            }

            /* Links with "cAsE" anywhere in the URL,
            with matching capitalization */
            a[href*="cAsE" s] {
                color: pink;
            }

            /* Links with "insensitive" anywhere in the URL,
            regardless of capitalization */
            a[href*="insensitive" i] {
                color: cyan;
            }

           /* Links with "example" anywhere in the URL */
            a[href*="example"] {
                background-color: silver;
            }

            /* Internal links, beginning with "#" */
            a[href^="#"] {
                background-color: gold;
            }

            a {
                color: blue;
            }

            ul > li {}  

            ul || li {}

            ul, li {}

            ul|li {}
            
            ul + li {}

            ul ~ li {}

```

### CODE REVIEW

**Review by jaytaph — APPROVED** [review]

File: src/css3/node.rs [hunk 1]
```diff
@@ -135,9 +166,31 @@ pub struct CDC;
...
+#[derive(PartialEq)]
+pub enum Combinator {
+    ChildCombinator,
+    ColumnCombinator,
+    SelectorListCombinator,
+    DescendantCombinator,
+    NamespaceSeparator,
+    NextSiblingCombinator,
+    SubsequentSiblingCombinator,
+}
```
Comment (emwalker) [inline src/css3/node.rs:178]: Interesting name.  I look forward to seeing what you do with this enum.

File: src/css3/tokenizer.rs [hunk 2]
```diff
@@ -91,13 +91,13 @@ mod test {
         let mut tokenizer = Tokenizer::default();
         tokenizer.init("123 -ident-test-1");
 
-        assert!(!tokenizer.is_eof());
-        assert!(tokenizer.has_more_tokens());
+        assert_eq!(tokenizer.is_eof(), false);
+        assert_eq!(tokenizer.has_more_tokens(), true);
```
Comment (emwalker) [inline src/css3/tokenizer.rs:95]: Can you run `make fix` on this PR?  I'm wondering whether these changes will go back to what was there before.

File: src/lib.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ extern crate alloc;
 
 pub mod api;
 #[allow(dead_code)]
-pub mod css;
+pub mod css3;
```
Comment (emwalker) [inline src/lib.rs:5]: Awesome.

**Review by emwalker — CHANGES_REQUESTED** [review]
Looks like one of the commits is failing checks.  Can you either squash the commits into a single commit and force push, or rebase and fix the checks on each commit?  This will help with `git bisect` in the future.

```sh 
$ make fix 
$ make test
```

**Review by emwalker — APPROVED** [review]

**Comment by blaumeise20** [comment]
What does `ul || li` do? Did I miss something here?

**Review by Kiyoshika — APPROVED** [review]

**Comment by blaumeise20** [comment]
> [Column Selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Column_combinator)

Oh, thanks. So we would be the first browser to support this? 😉

**Pull request merged**